### ISSUE TITLE
Support ordering operators (i.e. <, >, <=, >=) for `resource.Quantity` type in CRD XValidations

### ISSUE THREAD

**Issue by mfaizanse** [kubernetes-sigs/controller-tools#827]
The ordering operators (i.e. <, >, <=, >=) are not working on `resource.Quantity` types in [CRD XValidations](https://book.kubebuilder.io/reference/markers/crd-validation.html) (in k8s v1.26). They are just doing simple string comparisons. 

e.g.
```
// +kubebuilder:validation:XValidation:rule="self.a.memStorage.size < self.resources.limits.memory", message="memStorage.size should be less than resources.limits.memory."
```

We should add the operator overrides for `resource.Quantity`.

**Comment by jpbetz** [comment]
Does https://github.com/kubernetes/kubernetes/pull/118803 provide what is needed? This is available in 1.29+ I believe.

cc @alexzielenski

**Comment by sbueringer** [comment]
I think so, but let's see

@mfaizanse  WDYT?

/reopen
/remove-lifecycle rotten
